### PR TITLE
Change limit for role names to 100

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Role names can be 100 characters long now.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/requests/restaction/RoleActionImpl.java [hunk 1]
```diff
@@ -69,7 +69,8 @@ public Guild getGuild()
     @CheckReturnValue
     public RoleActionImpl setName(String name)
     {
-        this.name = name;
+        Checks.check(name == null || name.isEmpty() || name.length() <= 100, "Name may not be longer than 100 characters");
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/requests/restaction/RoleActionImpl.java]: This is not the same as how the RoleManager handles the Checks for name. Is there a reason that we want to convert `""` to `null`?

Seems like that isn't great for a user experience.

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/internal/requests/restaction/RoleActionImpl.java]: Creating roles without setting a name automatically sets the name to a default value, this doesn't work for the manager afaik

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/internal/requests/restaction/RoleActionImpl.java]: Ah let me rephrase. We shouldn't let `""` abstract to `null` behind the scenes. If someone is calling the `setName` method, it should be expected that they are providing a valid name, or providing `null` to reset a previous setter.

If they want to opt-in to the "no name provided, thus default random name" functionality, they need to either not call the function, or call it with the value that represents "no name provided" (`null`).

We should surface if an erroneous name was provided to help developer debug.
